src/clarity-angular/popover/popover.ts

```diff
@@ -45,33 +56,203 @@ export class Popover {
         // Browsers don't agree with what to do if some of these are not specified, so we set them all to be safe.
         element.style.position = POSITION_ABSOLUTE;
         element.style.top = 0;
+        element.style.bottom = "auto";
         element.style.left = 0;
+        element.style.right = "auto";
     }
 
-    public anchor(anchor: any, direction: Direction,
-                  {offsetX = 0, offsetY = 0, userAnchorParent = false}: PopoverOptions = {}) {
+    public anchor(
+        anchor: any, anchorAlign: Point, popoverAlign: Point,
+        {offsetX = 0, offsetY = 0, useAnchorParent = false}: PopoverOptions = {}) {
+
         // TODO: we are assuming here that the popover is inside or next to the anchor.
         // We'd need to go up the popover tree too otherwise
+
         this.preventScrolling(anchor);
-        if (userAnchorParent) {
+        if (useAnchorParent) {
             anchor = anchor.parentNode;
         }
+        // explicitly override anchor's style to static
+        anchor.style.position = "static";
+
+        let marginLeft = parseInt(getComputedStyle(this.element).marginLeft, 10);
+        let marginRight = parseInt(getComputedStyle(this.element).marginRight, 10);
+        let marginTop = parseInt(getComputedStyle(this.element).marginTop, 10);
+        let marginBottom = parseInt(getComputedStyle(this.element).marginBottom, 10);
+
```

Comment:
There are a lot of lookups for `this.element`, `anchorRect.*`, and `popoverRect.*` in `anchor()`.

Should we consider using a variable inside of `anchor` for some of these values? I think it would make any future refactor easier to handle and fewer lookups generally helps performance.